### PR TITLE
[AppService] Bugfix: Better error messages for webapp commands

### LINKED ISSUE #15202: Webapp:Improve the error message for az webapp up runtime auto detection
There are 2 things to fix in this bug:

## Improve formating:

Notice the difference between az webapp up error message formatting and the formating for az --versions command.

1) Multiple lines instead of runoff text make it more readable.
2) links are color coded to stand out

image

Here is my proposal:

image

- break the error message into multiple sentences with line breaks
- apply link formatting to learn more link

## improve the message body

The most common mistake is running this command from he wrong folder, this is covered in the learn more doc, however we should try and surface it a bit more, the suggestion here is to add a hint:

**Hint:** are you in the right folder?

### PR DESCRIPTION
**Pull request by calvinsID**
**Description**  
Fixes #15202 

**Testing Guide**  
Better error messages around the AppService commands

- run `az webapp up -n webappName` in a directory that isn't source code
Better error msg:
image

- run `az webapp up -n webappName` without administrator privileges in a directory in C:
Better error msg:
image

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Review by qwordy — APPROVED** [review]

**Comment by yungezz** [comment]
hi @houk-ms could you pls help to look at this PR also since it's error message refinement? thanks.

**Comment by houk-ms** [comment]
It seems not a common case to control error message style when raising an error in command group. what do you think @yungezz @qwordy

**Comment by yungezz** [comment]
> It seems not a common case to control error message style when raising an error in command group. what do you think @yungezz @qwordy

Agree. hi @calvinsID , Azure CLI has a common error message and style designing, for consistent user experience. It's not common to control color in separate commands specifically. Could you pls leverage existing Azure CLI error format? @houk-ms could share more detail.

File: src/azure-cli/azure/cli/command_modules/appservice/_create_util.py [hunk 1]
```diff
@@ -165,8 +172,11 @@ def get_lang_from_content(src_path, html=False):
         runtime_details_dict['file_loc'] = package_netcore_file
         runtime_details_dict['default_sku'] = 'F1'
     else:  # TODO: Update the doc when the detection logic gets updated
-        raise CLIError("Could not auto-detect the runtime stack of your app, "
-                       "see 'https://go.microsoft.com/fwlink/?linkid=2109470' for more information")
+        raise CLIError("Could not auto-detect the runtime stack of your app.\n"
+                       "HINT: Are you in the right folder?\n"
+                        "For more information, see "
+                        + Fore.BLUE + Style.BRIGHT + "'https://go.microsoft.com/fwlink/?linkid=2109470'"
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/appservice/_create_util.py:178]: as shared in another comment, pls try to use existing CLI error framework for this.

**Review by yungezz — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/_create_util.py (hunk 1 above)
Comment (calvinsID) [reply src/azure-cli/azure/cli/command_modules/appservice/_create_util.py]: Changed back to regular

**Comment by qwordy** [comment]
> 
> 
> It seems not a common case to control error message style when raising an error in command group. what do you think @yungezz @qwordy

Agree. Do we have a documentation of how to write error messages?

**Comment by calvinsID** [comment]
Hi @yungezz, I've changed the error msg styling back to normal

**Review by houk-ms — APPROVED** [review]

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
hi @qwordy could you pls help to take care of merging? thanks.

**Pull request merged**